regtest/bitcoind.js

```diff
@@ -0,0 +1,461 @@
+                      transactionData.push(txresponse.result.hex);
+                      finished();
+                    });
+                  },
+                  function(finished) {
+                    // Get the private key for each utxo
+                    client.dumpPrivKey(utxo.address, function(err, privresponse) {
+                      if (err) {
+                        throw err;
+                      }
+                      utxo.privateKeyWIF = privresponse.result;
+                      finished();
+                    });
+                  }
+                ], next);
+              }, function(err) {
+                if (err) {
+                  throw err;
+                }
+                done();
+              });
+            });
+          });
+        });
+      });
+    });
+  });
+
+  after(function(done) {
+    this.timeout(20000);
+    bitcoind.node.stopping = true;
+    bitcoind.stop(function(err, result) {
+      done();
+    });
+  });
+
+  describe('get blocks by hash', function() {
+
+    [0,1,2,3,5,6,7,8,9].forEach(function(i) {
+      it('generated block ' + i, function(done) {
+        bitcoind.getBlock(blockHashes[i], function(err, block) {
+          if (err) {
+            throw err;
+          }
+          should.exist(block);
+          block.hash.should.equal(blockHashes[i]);
+          done();
+        });
+      });
+    });
+  });
+
+  describe('get blocks as buffers', function() {
+    [0,1,2,3,5,6,7,8,9].forEach(function(i) {
+      it('generated block ' + i, function(done) {
+        bitcoind.getRawBlock(blockHashes[i], function(err, block) {
+          if (err) {
+            throw err;
+          }
+          should.exist(block);
+          (block instanceof Buffer).should.equal(true);
+          done();
+        });
+      });
+    });
+  });
+
+  describe('get errors as error instances', function() {
+    it('will wrap an rpc into a javascript error', function(done) {
+      bitcoind.client.getBlock(1000000000, function(err, response) {
+        var error = bitcoind._wrapRPCError(err);
+        (error instanceof Error).should.equal(true);
+        error.message.should.equal(err.message);
+        error.code.should.equal(err.code);
+        should.exist(error.stack);
+        done();
+      });
+    });
+  });
+
+  describe('get blocks by height', function() {
+
+    [0,1,2,3,4,5,6,7,8,9].forEach(function(i) {
+      it('generated block ' + i, function(done) {
+        // add the genesis block
+        var height = i + 1;
+        bitcoind.getBlock(i + 1, function(err, block) {
+          if (err) {
+            throw err;
+          }
+          should.exist(block);
+          block.hash.should.equal(blockHashes[i]);
+          done();
+        });
+      });
+    });
+
+    it('will get error with number greater than tip', function(done) {
+      bitcoind.getBlock(1000000000, function(err, response) {
+        should.exist(err);
```

Comment:
Added in: https://github.com/bitpay/bitcore-node/pull/422/commits/85a0c16eef58380ecb3ed2d1a00f3b23e0966604